dispatch/static/manager/src/js/components/EventEditor/index.js

```diff
@@ -59,17 +61,50 @@ const mapDispatchToProps = (dispatch) => {
     },
     deleteListItem: (token, eventId, next) => {
       dispatch(eventsActions.delete(token, eventId, next))
+    },
+    publishEvent(token, id) {
+      const form = new FormData()
+      form.append('is_published', true)
+      dispatch(eventsActions.save(token, id, form))
+    },
+    unpublishEvent(token, id) {
+      const form = new FormData()
+      form.append('is_published', false)
+      dispatch(eventsActions.save(token, id, form))
     }
   }
 }
 
 function EventEditorComponent(props) {
+
+  const publishButton = (
+    <AnchorButton onClick={() => props.publishEvent(props.token, props.listItem.id)}
+      intent={Intent.PRIMARY}
+      disabled={props.isNew} >
+      <span className='pt-icon-standard pt-icon-th'></span>
+      Publish
+    </AnchorButton>
+  )
+
+  const unpublishButton = (
+    <AnchorButton onClick={() => props.unpublishEvent(props.token, props.listItem.id)}
+      intent={Intent.PRIMARY}
+      disabled={props.isNew} >
+      <span className='pt-icon-standard pt-icon-th'></span>
+      Unpublish
+    </AnchorButton>
+  )
+
+  const event_is_published = props.entities.local && props.listItem.id
```

Comment:
Also, this could probably be moved into a helper outside of the component and called like this:

```js
extraButton={isPublished(props) ? unpublishButton : publishButton}
```